File: 프로그래머스/lv1/131128. 숫자 짝꿍/숫자 짝꿍.py
def solution(X, Y):
    answer = ''
    x_lst = [0] * 10
    y_lst = [0] * 10
    for i in range(len(X)) :
        x_lst[int(X[i])] += 1
    for i in range(len(Y)) :
        y_lst[int(Y[i])] += 1
    for i in range(9, -1, -1) :
        for j in range(min(x_lst[i], y_lst[i])) :
            answer += str(i)
    if len(answer) == 0 :
        answer = "-1"
    elif answer[0] == '0':
        answer = "0"
    return answer

File: 프로그래머스/lv1/131128. 숫자 짝꿍/숫자 짝꿍.cpp
#include <string>
#include <vector>

using namespace std;
#include <iostream>
string solution(string X, string Y) {
    string answer = "";
    vector<int> vx(10, 0);
    vector<int> vy(10, 0);
    for (int i = 0; i < X.length(); i++) vx[X[i] - '0']++;
    for (int i = 0; i < Y.length(); i++) vy[Y[i] - '0']++;
    vector<int> ans(10, 0);
    for (int i = 0; i < 10; i++) ans[i] = min(vx[i], vy[i]);
    for (int i = 9; i >= 0; i--) {
        if (ans[i] != 0) {
            for (int j = 0; j < ans[i]; j++) answer += to_string(i);
        }
    }
    return (answer.length() == 0) ? "-1" : (answer[0] == '0') ? "0" : answer;
}